srk/src/terminationLLRF.ml

```diff
@@ -0,0 +1,425 @@
+    coeff_x_list: a list of symbolic coefficients in the final linear terms
+    coeff_x_set: set of symbolic coefficients
+*)
+let build_system_of_ineq_for_non_inc srk cs ineqs dx_list coeff_x_list coeff_x_set =
+  (* first create the lambda symbols for each inequality and set that they are non-negative *)
+  let lambdas, f_with_non_neg_lambdas = 
+    BatList.fold_righti
+      (fun i ineq (lambdas, f) ->
+         let lambda_i_name = String.concat "_" ["lambda"; string_of_int i] in
+         let lambda_i = mk_const srk (mk_symbol srk ~name:lambda_i_name `TyReal) in
+         match ineq with
+         | `LeqZero _ ->
+           begin
+             let newf = mk_and srk [mk_leq srk (mk_zero srk) lambda_i;  f] in
+             (List.cons (lambda_i, lambda_i) lambdas, newf)
+           end
+         | `LtZero _ -> failwith "expecting non-strict ineqs"
+         | `EqZero _ -> 
+           begin
+             let lambda_ip_name = String.concat "_" ["lambda"; "neg"; string_of_int i] in
+             let lambda_ip = mk_const srk (mk_symbol srk ~name:lambda_ip_name `TyReal) in
+             let newf1 = mk_and srk [mk_leq srk (mk_zero srk) lambda_i; f] in
+             let newf2 = mk_and srk [mk_leq srk (mk_zero srk) lambda_ip; newf1] in
+             (List.cons (lambda_i, lambda_ip) lambdas, newf2)
+           end
+      )
+      ineqs
+      ([], mk_true srk)
+  in
+  (** then add the coefficient constraints to the system: for each variable,
+      linear combination of its coefficient within each inequality results in  
+      the final coefficients of this variable in the final linear term
+   *)
+  let answer =
+    BatList.fold_lefti
+      (fun formulas i sym -> 
+         let full_rhs =
+           BatList.fold_lefti 
+             (fun rhs i ineq ->
+                match ineq with
+                | `LeqZero t ->
+                  begin
+                    let lambda_i, _ = List.nth lambdas i in
+                    let coeff = get_coeff_of_symbol cs t sym in
+                    mk_add srk [rhs; mk_mul srk [lambda_i; mk_real srk coeff] ]
+                  end
+                | `LtZero _ -> failwith "expecting non-strict ineqs"
+                | `EqZero t -> 
+                  begin
+                    let coeff = get_coeff_of_symbol cs t sym in
+                    let lambda_i, lambda_ip = List.nth lambdas i in
+                    mk_add srk [rhs; 
+                                mk_mul srk [lambda_i; mk_real srk coeff];
+                                mk_mul srk [lambda_ip; mk_neg srk (mk_real srk coeff)] ]
+                  end
+             )
+             (mk_zero srk)
+             ineqs
+         in
+         let coeff_sym = List.nth coeff_x_list i in
+         let equ_for_this_var = mk_eq srk (mk_const srk coeff_sym) full_rhs in
+         mk_and srk [ equ_for_this_var ; formulas]
+      )
+      f_with_non_neg_lambdas
+      dx_list
+  in
+  logf "\nformula of lambdas on non-inc:\n%s\n\n" (Formula.show srk answer);
```

Comment:
Don't use show to pretty-print -- use pp.  show doesn't respect formatting constraints of surrounding context and it slows things done even when we don't end up printing to the terminal